pkg/network/common/egressip.go

```diff
@@ -177,10 +187,11 @@ func (eit *EgressIPTracker) UpdateHostSubnetEgress(hs *networkapi.HostSubnet) {
 
 	node := eit.nodesByNodeIP[hs.HostIP]
```

Comment:
Do we have an issue in this case?
- Got add HostSubnet event, h1 with host IP ip1 and egress IP e1
- So we populated nodesByNodeIP[ip1] = { nodeIP: ip1, requestedIPs: [ e1 ], ... }
- Got modified HostSubnet event for h1 and host IP changed from ip1 to ip2
- We populate nodesByNodeIP[ip2]  = { nodeIP: ip2, requestedIPs: [ e1 ] ...}
   So now egress IP e1 is used by ip1 and ip2 according to the map even though ip1 is no longer valid.
   We need to delete obsoleted nodesbyNodeIP[ip1] and we could do that by keeping track of node UID --> node IP map?

Reply:
Doh. Yes, that applies to the existing "semi-automatic" code too though, so that's separate from this new feature. I'll do another PR to fix that after this.